src/types.rs

```diff
@@ -1620,20 +2075,102 @@ pub struct RuleContext {
 /// A first-order term rewriting system.
 #[derive(Debug, Hash, Clone, PartialEq, Eq)]
 pub struct TRS {
+    /// # Examples
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Rule, parse_rule, TRS, parse_trs};
+    /// let mut sig = Signature::default();
+    /// 
+    /// // Constructing a TRS manually 
+    /// let r0 = parse_rule(&mut sig, "A(x_) = A(B)").expect("parse of A(x_) = A(B)");
+    /// let r1 = parse_rule(&mut sig, "B = C | D").expect("parse of B = C | D");
+    /// let r2 = parse_rule(&mut sig, "E(F) = G").expect("parse of E(F) = G");
+    /// 
+    /// let t = TRS::new(vec![r0, r1, r2]);
+    ///
+    /// // Constructing a TRS using the parser
+    /// let t = parse_trs(&mut sig, "A(x_) = A(B);\nB = C | D;\nE(F) = G;")
+    ///     .expect("parse of A(x_) = A(B); B = C | D; E(F) = G;");
+    /// 
+    /// // or alternatively for readability of the TRS
+    /// let t = parse_trs(&mut sig, 
+    /// "A(x_) = A(B);
+    /// B = C | D;
+    /// E(F) = G;").expect("parse of A(x_) = A(B); B = C | D; E(F) = G;");
+    /// ```
     is_deterministic: bool,
     pub rules: Vec<Rule>,
 }
 impl TRS {
-    /// Constructs a term rewriting system from a list of rules.
-    pub fn new(rules: Vec<Rule>) -> TRS {
-        TRS {
-            rules,
-            is_deterministic: false,
-        }
+    /// Constructs a [`Term Rewriting System`] from a list of `Rule`s.
+    ///
+    /// [`Term Rewriting System`]: https://en.wikipedia.ord/wiki/Rewriting#Term_rewriting_systems
+    ///
+    /// # Examples 
+    ///
+    /// ```
+    /// # use term_rewriting::{Signature, Rule, parse_rule, TRS};
+    /// let mut sig = Signature::default();
+    /// 
+    /// let r0 = parse_rule(&mut sig, "A = B").expect("parse of A = B");
+    /// let r1 = parse_rule(&mut sig, "C(x_) = x_").expect("parse of C(x_) = x_");
+    /// let r2 = parse_rule(&mut sig, "D(y_) = D(E)").expect("parse of D(y_) = D(E)");
+    /// let new_trs = TRS::new(vec![r0, r1, r2]);
+    ///
+    /// assert_eq!(new_trs.display(&sig), 
+    /// "A = B;
+    /// C(x_) = x_;
+    /// D(y_) = D(E);"
+    /// );
+    /// ```
+    pub fn new(rules: Vec<Rule>) -> TRS {
+        TRS {
+            rules,
+            is_deterministic: false,
+        }
     }
     /// Make `self` deterministic and restrict it to be so until further notice.
     ///
-    /// Return `false` if `self` was changed, otherwise `false`.
+    /// Return `true` if `self` was changed, otherwise `false`.
+    ///
+    /// See [`Deterministic System`] for more information.
```

Comment:
Go ahead and move this link to the word `deterministic` in the previous sentence so we can delete this sentence.